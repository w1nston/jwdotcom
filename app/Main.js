import React from 'react';
import Router from 'react-router/BrowserRouter';
import Link from 'react-router/Link';
import Match from 'react-router/Match';
import Home from './pages/home/components/Home';
import Blog from './pages/blog/components/Blog';

export default function Main() {
  return (
    <Router>
      <div>
        <section className="navigation-container">
          <img
            className="navigation-container__jw-logo"
            src="public/assets/images/jwlogo.svg" />
          <nav className="navigation-container__nav">
            <Link to="/">Home</Link>
            <Link to="/blog">Blog</Link>
          </nav>
        </section>
        <section className="main-content-container">
          <Match exactly pattern="/" component={Home} />
          <Match pattern="/blog" component={Blog} />
        </section>
      </div>
    </Router>
  );
}
