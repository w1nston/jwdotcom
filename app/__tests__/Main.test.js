import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Main from '../Main';

test('that the Main routes are rendered correctly', () => {
  const component = shallow(<Main />);
  expect(shallowToJson(component)).toMatchSnapshot();
});
