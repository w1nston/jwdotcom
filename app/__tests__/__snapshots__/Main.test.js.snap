exports[`test that the Main routes are rendered correctly 1`] = `
<BrowserRouter>
  <div>
    <section
      className="navigation-container">
      <img
        className="navigation-container__jw-logo"
        src="public/assets/images/jwlogo.svg" />
      <nav
        className="navigation-container__nav">
        <Link
          activeClassName=""
          activeOnlyWhenExact={false}
          activeStyle={Object {}}
          className=""
          isActive={[Function]}
          replace={false}
          style={Object {}}
          to="/">
          Home
        </Link>
        <Link
          activeClassName=""
          activeOnlyWhenExact={false}
          activeStyle={Object {}}
          className=""
          isActive={[Function]}
          replace={false}
          style={Object {}}
          to="/blog">
          Blog
        </Link>
      </nav>
    </section>
    <section
      className="main-content-container">
      <Match
        component={[Function]}
        exactly={true}
        pattern="/" />
      <Match
        component={[Function]}
        exactly={false}
        pattern="/blog" />
    </section>
  </div>
</BrowserRouter>
`;
